code/modules/fish/fishtank.dm

```diff
@@ -332,29 +335,41 @@
 	for(var/key in fish_types) // Then populate the list
 		var/datum/fish/fish_type = key
 		var/count = length(fish_types[key])
-		fish_types_input += list("[initial(fish_type.fish_name)][count > 1 ? " (x[count])" : ""]" = fish_types[key])
+		fish_types_input += list("[initial(fish_type.fish_name)][count > 1 ? " (x[count])" : ""]" = key)
```

Comment:
```suggestion
		var/fish_name = "[initial(fish_type.fish_name)][count > 1 ? " (x[count])" : ""]"
		fish_types_input[fish_name] = fish_type
```
This should be the same. And hopefully a bit less confusing. Be sure to test it just in case